docs/rules/key-spacing.md

```diff
@@ -1,115 +1,132 @@
 
 ## Rule Details
 
-This rule will warn when spacing in properties does not match the specified options. In the case of long lines, it is acceptable to add a new line wherever whitespace is allowed. There are three modes:
+This rule enforces consistent spacing between keys and values in object literal properties. In the case of long lines, it is acceptable to add a new line wherever whitespace is allowed.
 
 ## Options
 
-Use the `beforeColon`, `afterColon` and `mode` options to enforce having one space or zero spaces on each side, using `true` or `false`, respectively. The default is no whitespace between the key and the colon and one space between the colon and the value.
+This rule has an object option:
 
-`mode` option can be either `"strict"` or `"minimum"` and defaults to `"strict"`. In `strict` mode, it enforces exactly 1 space before or after the colon where as in `minimum` mode, it enforces at least 1 space but more are okay.
+* `"beforeColon"` (default `false`) enforces consistent spacing style between the key and the colon in object literals
+* `"afterColon"` (default `true`) enforces consistent spacing style between the colon and the value in object literals
+* `"mode": strict` (default) enforces exactly one space before or after colons in object literals
+* `"mode": minimum` enforces one or more spaces before or after colons in object literals
+* `"align": "value"` enforces vertical alignment of values in object literals
+* `"align": "colon"` enforces vertical alignment of both colons and values in object literals.
+* `"singleLine"` enforces a spacing style for single-line object literals
+* `"multiLine"` enforces a spacing style for multi-line object literals
 
-The following patterns are considered valid:
+Please note that you can either use the top-level options or the grouped options (`singleLine` and `multiLine`) but not both.
 
-```js
-// DEFAULT
-/*eslint key-spacing: ["error", {"beforeColon": false, "afterColon": true}]*/
+### beforeColon
 
-var obj = { "foo": (42) };
+Examples of **incorrect** code for this rule with the default `{ "beforeColon": false }` option:
+
+```js
+/*eslint key-spacing: ["error", { "beforeColon": false }]*/
 
-foo = { thisLineWouldBeTooLong:
-    soUseAnotherLine };
+var obj = { "foo" : (42) };
 ```
 
+Examples of **correct** code for this rule with the default `{ "beforeColon": false }` option:
+
 ```js
-/*eslint key-spacing: ["error", {"beforeColon": true, "afterColon": false}]*/
+/*eslint key-spacing: ["error", { "beforeColon": false }]*/
 
-call({
-    foobar :42,
-    bat :(2 * 2)
-});
+var obj = { "foo": (42) };
 ```
 
+Examples of **incorrect** code for this rule with the `{ "beforeColon": true }` option:
+
 ```js
-/*eslint key-spacing: ["error", {"beforeColon": true, "afterColon": false, "mode": "minimum"}]*/
+/*eslint key-spacing: ["error", { "beforeColon": true }]*/
 
-call({
-    foobar   :42,
-    bat :(2 * 2)
-});
+var obj = { "foo":(42) };
```

Comment:
Since the eslint comment does not change the default value `"afterColon": true`, it is probably safest to keep consistent one space after for all 4 examples under **beforeColon**.